crux-bench/src/crux/bench/main.clj

```diff
@@ -3,11 +3,25 @@
             [crux.bench.ts-weather :as weather]
             [crux.bench.ts-devices :as devices]
             [clj-http.client :as client]
-            [clojure.tools.logging :as log]))
+            [clojure.tools.cli :as cli]
+            [crux.bench.watdiv :as watdiv]))
 
-(defn -main []
-  (bench/post-to-slack (format "*Starting Benchmark*, Crux Version: %s, Commit Hash: %s\n"
-                               bench/crux-version bench/commit-hash))
-  (bench/with-node [node]
-    (devices/run-devices-bench node)
-    (weather/run-weather-bench node)))
+(def cli-options
+  [["-nc" "--no-crux" "Don't run Crux benchmark"]
```

Comment:
nice idea, but this is a little unusual (to me, at least) - I can't think of another example of using flags to alter the behaviour of the application so much - we've essentially got three different applications here now. also, if you're going to use `-`, stick to a single character, `--` for long options, (as you've done). the `java` CLI is very much the annoying exception here rather than the rule!

I'd be tempted to have these as different Clojure main namespaces, which you can then specify using the docker CMD

Reply:
considering we are already specifying the main directly anyway - yeah, I like that idea far more.